packages/fxa-profile-server/lib/routes/profile.js

```diff
@@ -84,11 +89,26 @@ module.exports = {
     }
 
     return server.methods.profileCache.get(req).then((response) => {
-      let result = response.value;
-      if (response.value.result) {
-        // safety check for several entries that were wrapped in an extra result.
-        // this can be safely removed after train-165 is deployed.
-        result = response.value.result;
+      const result = response.value;
+      if (result.avatarDefault || result.avatar.startsWith(monogramUrl)) {
+        const displayName = result.displayName;
+        let avatarUrl = result.avatar;
+        if (displayName && ALPHANUMERIC.test(displayName)) {
+          avatarUrl = `${monogramUrl}/v1/avatar/${displayName[0]}`;
+          result.avatarDefault = false;
+        } else if (ALPHANUMERIC.test(result.email)) {
+          avatarUrl = `${monogramUrl}/v1/avatar/${result.email[0]}`;
+          result.avatarDefault = false;
+        }
+        if (avatarUrl !== result.avatar) {
+          result.avatar = avatarUrl;
+          db.addAvatar(
```

Comment:
@vbudhram I think this is a reasonable compromise for making `/avatar` return the monogram. I was hoping to not need to use the db, but the alternative means we need to fetch email from auth-server inside `/avatar`. This seems less gross. It relies on calling `/profile` once before `/avatar` to set it, but in normal cases I think that'll always be the case.

Reply:
Seems fair, I think it might be worthwhile to leave a comment about this here too.